### DISCUSSION TITLE
CP-SAT solver fails to prove infeasibility

### DISCUSSION THREAD

**Discussion by Mapiarz** [google/or-tools#3338] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
Versions: pypi ortools-9.0.9048 and ortools-9.3.10497
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT

**What operating system (Linux, Windows, ...) and version?**
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.6 LTS
Release:	18.04
Codename:	bionic

**What did you do?**
We have upgraded from `ortools-9.0.9048` to `ortools-9.3.10497` and discovered that the new version fails to prove infeasibility of **some** of our models. 

**What did you expect to see**
On 9.0, solver behaves as expected: proves infeasibility within a few seconds:
```
Statistics
  - status          : INFEASIBLE
  - conflicts       : 2
  - branches        : 6751
  - wall time       : 4.776481 s
```

**What did you see instead?**
On 9.3, solver fails to prove infeasibility and ultimately returns UNKNOWN status (after reaching time limit):
```
Statistics
  - status          : UNKNOWN
  - conflicts       : 15319
  - branches        : 143430
  - wall time       : 2512.561640 s
```

**Make sure you include information that can help us debug (full error message, model Proto).**
See below for an example model that should allow you reproduce the issue across the 2 versions:
- bin: https://www.dropbox.com/s/all9lagqj3c1j6q/model_proto.bin?dl=1
- txt: https://www.dropbox.com/s/y1bbwcqzwumjkec/model_proto.txt?dl=1

**Extra note**
We have also empirically detected a notable degradation in solving performance between 9.0 and 9.3.

**Comment by lperron** [comment]
can you attach the proto from the 9.3 release (the format has changed in between) ?

**Reply by Mapiarz** [reply]
@lperron Here is the model generated using 9.3 release:
- bin: https://www.dropbox.com/s/2cxwmr8xi70gfyi/model_proto_9_3.bin?dl=1
- txt: https://www.dropbox.com/s/k2wrwq1jqmrfz31/model_proto_9_3.txt?dl=1

**Comment by lperron** [comment, marked as answer]
On the main branch:

 CpSolverResponse summary:
status: INFEASIBLE
objective: NA
best_bound: NA
booleans: 3380
conflicts: 0
branches: 6760
propagations: 149671
integer_propagations: 224300
restarts: 6761
lp_iterations: 3326
walltime: 1.46383
usertime: 9.22279
deterministic_time: 2.59648
gap_integral: 0

**Reply by Mapiarz** [reply]
@lperron Thanks. Glad this is resolved on `master`.

Any timeline on next release that will include this (whatever it is) fix?

**Comment by lperron** [comment]
This month for sure. 1-2 weeks I would say.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
